packages/wrangler/src/pages/dev.ts

```diff
@@ -402,7 +402,19 @@ export const Handler = async (args: PagesDevArguments) => {
 			}
 		});
 	} else if (usingWorkerScript) {
+		/*
+		 *
+		 */
+		// always watch _worker.js
+		const watcher = watch([workerScriptPath], {
+			persistent: true,
+			ignoreInitial: true,
+		});
+		let currentBundleDependencies: string[] = [];
+		let watchedBundleDependencies: string[] = [];
```

Comment:
Is there a reason to have these as arrays given the functions dir implementation uses `Set`s?

Reply:
this was me trying to simplify what I had in the `Functions` implementation.  That one is a tad more verbose because of the `if...else` clauses, as opposed to this one, which does a lot of array ops chaining, which is why sets were no longer necessary. 

I hesitated writing the code this, because IMO the `Functions` one is a tad more readable...but then again this whole branching of `isUsingWorkerjs` | `isUsingFunctions` | `isUsingWorkerjsDirectory` is getting harder and harder to read :/

Reply:
refactored Functions to be in line with this PR here: https://github.com/cloudflare/workers-sdk/pull/6150/commits/536836fcd5791a58c3f9b7297b36eb391cdd528c